scripts/gha/utils.py

```diff
@@ -218,3 +218,45 @@ def clean_vcpkg_temp_data():
   for directory_to_remove in directories_to_remove:
     abspath = os.path.join(vcpkg_root_dir_path, directory_to_remove)
     delete_directory(abspath)
+
+
+def install_x86_support_libraries(gha_build=False):
+  """Install support libraries needed to build x86 on x86_64 hosts.
+
+  Args:
+    gha_build: Pass in True if running on a GitHub runner; this will activate
+               workarounds that might be undesirable on a personal system (e.g.
+               downgrading Ubuntu packages).
+  """
+  if is_linux_os():
+    packages = ['gcc-multilib', 'g++-multilib', 'libglib2.0-dev:i386',
+                'libsecret-1-dev:i386', 'libpthread-stubs0-dev:i386',
+                'libssl-dev:i386']
+    if gha_build:
+      # Workaround for GitHub runners, which have an incompatibility between the
+      # 64-bit and 32-bit versions of the Ubuntu package libpcre2-8-0. Downgrade
+      # the installed 64-bit version of the library to get around this issue.
+      # This will presumably be fixed in a future Ubuntu update. (If you remove
+      # it, remove the workaround further down this function as well.)
+      packages = ['--allow-downgrades'] + packages + ['libpcre2-8-0=10.34-7']
+
+    # First check if these packages exist on the machine already
+    devnull = open(os.devnull, "w")
```

Comment:
drive-by comment: use a `with` block to ensure `devnull` gets closed even if `subprocess.run()` raises an exception.

i.e.

```
with open(os.devnull, "w") as devnull:
  subprocess.run(...)
```